test-case/check-keyword-detection-in-D0i3.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+set -e
+
+##
+## Case Name: check-keyword-detection-in-D0i3
+## Preconditions:
+##    N/A
+## Description:
+##    Run arecord on kwd pipeline first, wait DSP enters to D0i3, then play a
+##    a special sine wav(997 Hz sine wav which generated by wav tool) to trigger
+##    kwd and then use wavtool to do the analysis.
+## Case step:
+##    1. use wavtool to generate the test wav file.
+##    2. run kwd first and wait DSP enters to D0i3.
+##    3. check the DSP status.
+##    4. play the test wav through the USB card to trigger kwd.
+##    5. after the test, use wavetool to analyze the recorded wav.
+## Expect result:
+##    no errors and the captured data should match with the orignal one.
+##
+
+libdir=$(dirname "${BASH_SOURCE[0]}")
+# shellcheck source=case-lib/lib.sh
+source "$libdir"/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'              OPT_DESC_lst['t']='tplg file, default value is env TPLG: $''TPLG'
+OPT_PARM_lst['t']=1                  OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['s']='sof-logger'        OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0                  OPT_VALUE_lst['s']=1
+
+OPT_OPT_lst['p']='preamble-time'     OPT_DESC_lst['p']='key phrase preamble length'
+OPT_PARM_lst['p']=1                  OPT_VALUE_lst['p']=2100
+
+OPT_OPT_lst['s']='history-depth'     OPT_DESC_lst['s']='draining size'
+OPT_PARM_lst['s']=1                  OPT_VALUE_lst['s']=2100
+
+OPT_OPT_lst['b']='buffer'            OPT_DESC_lst['b']='buffer size'
+OPT_PARM_lst['b']=1                  OPT_VALUE_lst['b']=67200
+
+OPT_OPT_lst['l']='loop'              OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1                  OPT_VALUE_lst['l']=1
+
+OPT_OPT_lst['d']='duration'          OPT_DESC_lst['d']='interrupt kwd pipeline in # seconds'
+OPT_PARM_lst['d']=1                  OPT_VALUE_lst['d']=10
+
+func_opt_parse_option "$@"
+
+tplg=${OPT_VALUE_lst['t']}
+loop_cnt=${OPT_VALUE_lst['l']}
+buffer_size=${OPT_VALUE_lst['b']}
+history_depth=${OPT_VALUE_lst['s']}
+preamble_time=${OPT_VALUE_lst['p']}
+duration=${OPT_VALUE_lst['d']}
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+func_pipeline_export "$tplg" "kpbm:any"
+func_lib_setup_kernel_last_line
+
+if test "$PIPELINE_COUNT" != "1"; then
+    die "detected $PIPELINE_COUNT wov pipeline(s) from topology, but 1 is needed"
```

Comment:
Once again, the negation is not necessary:
```shell
test "$PIPELINE_COUNT" = "1" || 
     die ...
```
This reads like plain English and looks like `assert(PIPELINE_COUNT == 1)` in C which is exactly what it does.